babel/dates.py

```diff
@@ -1453,10 +1452,44 @@ def format_day_of_year(self, num):
     def format_day_of_week_in_month(self):
         return '%d' % ((self.value.day - 1) // 7 + 1)
 
-    def format_period(self, char):
-        period = {0: 'am', 1: 'pm'}[int(self.value.hour >= 12)]
-        for width in ('wide', 'narrow', 'abbreviated'):
-            period_names = get_period_names(context='format', width=width, locale=self.locale)
+    def format_period(self, char, num):
+        """
+        Return period from parsed datetime according to format pattern.
+
+        >>> format = DateTimeFormat(time(19, 42), 'en_US')
+        >>> format.format_period('a', 1)
+        u'PM'
+
+        >>> format.format_period('b', 1)
+        u'evening'
+
+        >>> format.format_period('B', 1)
+        u'in the evening'
+
+        >>> format = DateTimeFormat(datetime(2022, 4, 28, 6, 27), 'zh_Hant')
+        >>> format.format_period('a', 1)
+        u'上午'
+
+        >>> format.format_period('b', 1)
+        u'清晨'
+
+        >>> format.format_period('B', 1)
+        u'清晨'
+
+        :param char: pattern format character ('a', 'b', 'B')
+        :param num: count of format character
+
+        """
+        widths = [{3: 'abbreviated', 4: 'wide', 5: 'narrow'}[max(3, num)],
+                  'wide', 'narrow', 'abbreviated']
+        if char == 'a':
+            period = {0: 'am', 1: 'pm'}[int(self.value.hour >= 12)]
```

Comment:
I know this was from the original code, but now that we're touching this segment, this should probably be changed to the more obvious
```suggestion
            period = 'pm' if int(self.value.hour >= 12) else 'am'
```

Reply:
```suggestion
            period = 'pm' if self.value.hour >= 12 else 'am'
```
I think `int(...)` is not needed if changing the code.